### PR TITLE
Add nutanix-specific customizaitons to machines

### PR DESCRIPTION
**Pull request by thunderboltsid**
Add nutanix platform customizations in pkg/assets/machines.

### CODE REVIEW

**Comment by rvanderp3** [comment]
Please separate vendor changes into their own commits, to make reviewing and perusing history easier.

File: pkg/asset/machines/worker.go [hunk 1]
```diff
@@ -171,6 +175,17 @@ func defaultPowerVSMachinePoolPlatform() powervstypes.MachinePool {
 	}
 }
 
+func defaultNutanixMachinePoolPlatform() nutanixtypes.MachinePool {
+	return nutanixtypes.MachinePool{
+		NumCPUs:           2,
+		NumCoresPerSocket: 2,
+		MemoryMiB:         8192,
+		OSDisk: nutanixtypes.OSDisk{
+			DiskSizeMiB: 120 * 1024,
```
Comment (rvanderp3) [inline pkg/asset/machines/worker.go:184]: Other platforms derive their volume size from `decimalRootVolumeSize`:
`DiskSizeMiB: decimalRootVolumeSize * 1024`

Comment (thunderboltsid) [reply pkg/asset/machines/worker.go]: Oh, nice. Will fix that.

File: pkg/types/nutanix/helpers.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/google/uuid"
+	"github.com/kdomanski/iso9660"
+	"github.com/pkg/errors"
+	nutanixClientV3 "github.com/terraform-providers/terraform-provider-nutanix/client/v3"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+const (
+	diskLabel        = "config-2"
```
Comment (rvanderp3) [inline pkg/types/nutanix/helpers.go:18]: can a more descriptive label be provided?

File: pkg/asset/machines/worker.go [hunk 3]
```diff
@@ -520,6 +535,20 @@ func (w *Worker) Generate(dependencies asset.Parents) error {
...
+		case nutanixtypes.Name:
+			mpool := defaultNutanixMachinePoolPlatform()
+			mpool.Set(ic.Platform.Nutanix.DefaultMachinePlatform)
+			mpool.Set(pool.Platform.Nutanix)
+			pool.Platform.Nutanix = &mpool
+			imageName := nutanixtypes.RHCOSImageName(clusterID.InfraID)
+
+			sets, err := nutanix.MachineSets(clusterID.InfraID, ic, &pool, imageName, "worker", workerUserDataSecretName)
+			if err != nil {
+				return errors.Wrap(err, "failed to create worker machine objects")
```
Comment (rvanderp3) [inline pkg/asset/machines/worker.go:547]: i think `machine objects` should be `machinesets`

**Review by rvanderp3 — COMMENTED** [review]
Other than a nit, lgtm

File: pkg/asset/machines/worker.go (hunk 3 above)
Comment (thunderboltsid) [reply pkg/asset/machines/worker.go:547]: I concur with the terminology, however, I kept it same for consistency across providers.

File: pkg/asset/machines/master.go [hunk 4]
```diff
@@ -19,6 +19,8 @@ import (
 	libvirtprovider "github.com/openshift/cluster-api-provider-libvirt/pkg/apis/libvirtproviderconfig/v1beta1"
 	ovirtproviderapi "github.com/openshift/cluster-api-provider-ovirt/pkg/apis"
 	ovirtprovider "github.com/openshift/cluster-api-provider-ovirt/pkg/apis/ovirtprovider/v1beta1"
+	nutanixapi "github.com/openshift/machine-api-provider-nutanix/pkg/apis"
+	nutanixprovider "github.com/openshift/machine-api-provider-nutanix/pkg/apis/nutanixprovider/v1beta1"
```
Comment (staebler) [inline pkg/asset/machines/master.go:23]: Can we get these API types from github.com/openshift/api/machine/v1 instead? That keeps the installer from having to pull in dependencies like github.com/terraform-providers/terraform-provider-nutanix.

Comment (thunderboltsid) [reply pkg/asset/machines/master.go:23]: @staebler We will be doing that in a separate PR. Yanhua from our side is currently working on that PR.

Comment (thunderboltsid) [reply pkg/asset/machines/master.go:23]: In general, our perspective on this is that let’s get the existing stack merged. Some of the changes that require moving the dependency source over from MAPI Provider to API are better done all at once in a subsequent PR.

Comment (staebler) [reply pkg/asset/machines/master.go:23]: Help me to understand the difference between the types in openshift/api and openshift/machine-api-provider-nutanix, currently. Is what is in openshift/api what we are going to use for the release? Are we blocked by the changes that need to be made to the nutanix machine-api operator to support using the types in openshift/api?

Comment (thunderboltsid) [reply pkg/asset/machines/master.go:23]: @staebler AFAIK, the machine types are being ported from MAPI provider to API by Yanhua.

**Pull request merged**